lib/datasource/index.spec.ts

```diff
@@ -13,19 +13,33 @@ import { GalaxyDatasource } from './galaxy';
 import * as datasourceGithubTags from './github-tags';
 import * as datasourceMaven from './maven';
 import * as datasourceNpm from './npm';
-import * as datasourcePackagist from './packagist';
+import { PackagistDatasource } from './packagist';
 import type { DatasourceApi } from './types';
 import * as datasource from '.';
 
 jest.mock('./docker');
 jest.mock('./maven');
 jest.mock('./npm');
-jest.mock('./packagist');
+jest.mock('./packagist', () => ({
+  __esModule: true,
+  PackagistDatasource: jest.fn(() => {
+    const { PackagistDatasource: ActualPackagistDatasource } =
+      jest.requireActual('./packagist');
+    return Object.assign(Object.create(ActualPackagistDatasource.prototype), {
```

Comment:
Looks pretty weird. Why do we need the assignment? Can't we extend the class and set the override via datasources map?

Reply:
Yeah, let me give that a go. I need to instantiate a class, or a subtype of the class somewhere, so that it is an instance of `Datasource`.

Reply:
I am very glad you asked me to rewrite this. I'm much happier with the mock now. Thank you!